script/upgrades/eXOF/eXOFChecks.swap.sol

```diff
@@ -0,0 +1,233 @@
+
+    testAndPerformConstantProductSwap(
+      exchangeID,
+      trader,
+      tokenIn,
+      tokenOut,
+      amountIn,
+      config.eXOFCelo.referenceRateFeedID
+    );
+
+    console.log("🟢 eXOF -> CELO swap successful 🚀");
+  }
+
+  function swapBridgedEUROCtoEXOF(eXOFConfig.eXOF memory config) internal {
+    bytes32 exchangeID = getExchangeId(
+      config.eXOFEUROC.asset0,
+      config.eXOFEUROC.asset1,
+      config.eXOFEUROC.isConstantSum
+    );
+
+    address trader = vm.addr(4);
+    address tokenIn = bridgedEUROC;
+    address tokenOut = eXOF;
+    uint256 amountIn = 10e6;
+
+    // Mint some EUROC to trader
+    deal(tokenIn, trader, amountIn);
+
+    testAndPerformConstantSumSwap(
+      exchangeID,
+      trader,
+      tokenIn,
+      tokenOut,
+      amountIn,
+      config.eXOFEUROC.referenceRateFeedID,
+      true
+    );
+
+    console.log("🟢 bridgedEUROC -> eXOF swap successful 🚀");
+  }
+
+  function swapEXOFtoBridgedEUROC(eXOFConfig.eXOF memory config) internal {
+    bytes32 exchangeID = getExchangeId(
+      config.eXOFEUROC.asset0,
+      config.eXOFEUROC.asset1,
+      config.eXOFEUROC.isConstantSum
+    );
+
+    address trader = vm.addr(4);
+    address tokenIn = eXOF;
+    address tokenOut = bridgedEUROC;
+    uint256 amountIn = 100e18;
+
+    testAndPerformConstantSumSwap(
+      exchangeID,
+      trader,
+      tokenIn,
+      tokenOut,
+      amountIn,
+      config.eXOFEUROC.referenceRateFeedID,
+      false
+    );
+
+    console.log("🟢 eXOF -> bridgedEUROC swap successful 🚀");
+  }
+
+  // *** Helper Functions *** //
+
+  function testAndPerformConstantProductSwap(
+    bytes32 exchangeID,
+    address trader,
+    address tokenIn,
+    address tokenOut,
+    uint256 amountIn,
+    address rateFeedID
+  ) internal {
+    uint256 amountOut = Broker(broker).getAmountOut(biPoolManagerProxy, exchangeID, tokenIn, tokenOut, amountIn);
+    IBiPoolManager.PoolExchange memory pool = BiPoolManager(biPoolManagerProxy).getPoolExchange(exchangeID);
+
+    (uint256 numerator, uint256 denominator) = SortedOracles(sortedOraclesProxy).medianRate(rateFeedID);
+    FixidityLib.Fraction memory rate = FixidityLib.newFixedFraction(numerator, denominator);
+
+    FixidityLib.Fraction memory amountInAfterSpread = FixidityLib.newFixed(amountIn).multiply(
+      FixidityLib.newFixedFraction(9950, 10000)
+    );
+
+    uint256 estimatedAmountOut;
+    if (tokenIn == pool.asset0) {
+      estimatedAmountOut = amountInAfterSpread.divide(rate).fromFixed();
+    } else {
+      estimatedAmountOut = amountInAfterSpread.multiply(rate).fromFixed();
+    }
+
+    FixidityLib.Fraction memory maxTolerance = FixidityLib.newFixedFraction(25, 1000);
+    uint256 threshold = FixidityLib.newFixed(estimatedAmountOut).multiply(maxTolerance).fromFixed();
+
+    assertApproxEq(amountOut, estimatedAmountOut, threshold);
+    doSwapIn(exchangeID, trader, tokenIn, tokenOut, amountIn, amountOut);
+  }
+
+  function testAndPerformConstantSumSwap(
```

Comment:
In MU03 we refactored this function and simplified it I would suggest using that one. If not I would rename the USDC part to EUROC since we only perform Constant sum EUROC swaps here